include/SH3/graphics/msbmp.hpp

```diff
@@ -0,0 +1,130 @@
+        std::uint32_t pix_offset;   /**< Offset to beginning of image data */
+    };
+
+    /**
+     * Bitmap information header. Contains information about the actual bitmap (such as width and height)
+     */
+    struct bmp_info_header
+    {
+        std::uint32_t header_size;  /**< Size of the header following this one. Used to determine which one to use. */
+        std::int32_t width;         /**< Width of this image. For some reason it's a SIGNED 32-bit integer.... */
+        std::int32_t height;        /**< Height of this image. */
+        std::uint16_t colour_plane; /**< Number of colour planes. Should be 1. */
+        std::uint16_t bpp;          /**< Number of bits per pixel (for us, should be 8 or 24) */
+        std::uint32_t compression;  /**< Compression method used. THIS SHOULD ALWAYS BE 0 FOR UNCOMPRESSED!!!! */
+        std::uint32_t image_size;   /**< Size of this image in bytes */
+        std::int32_t x_res;         /**< Horizontal resolution in pixels per metre */
+        std::int32_t y_res;         /**< Vertical resolution in pixels per metre */
+        std::uint32_t palette_size; /**< Number of colours in the colour palette */
+        std::uint32_t imp_colours;  /**< Number of important colours used. 0 when every colour is important. Usually ignored */
+    };
+
+    #pragma pack(pop)
+
+    /**@}*/
+
+    /**
+     *  Describes a Microsoft bitmap file, which can be loaded and sent to the graphics card.
+     *  Note that all bmp files are stored on disk PHYSICALLY, therefore do not require the use
+     *  of the virtual filesystem.
+     */
+    struct msbmp final
+    {
+    public:
+        msbmp(const std::string& path){Load(path);}
+        ~msbmp(){}
+
+        enum class result
+        {
+            SUCCESS,
+            BAD_LOAD
+        };
+
+        struct load_result : public error<result>
+        {
+        public:
+            std::string message() const;
+        };
+
+        /**
+         *  Load all image data
+         *
+         *  @param path Physical path to the file (bmps are NOT contained in an arc section)
+         *
+         *  @return @ref load_result indicating whether or not the load failed.
+         */
+        load_result Load(const std::string& path);
+
+        /**
+         * Draw this bitmap to the framebuffer
+         */
+        void Draw();
+
+        /**
+         *  Get the width of this texture
+         */
+        std::int32_t GetWidth() const {return width;}
+
+        /**
+         *  Get the height of this texture
+         */
+        std::int32_t GetHeight() const {return height;}
+
+    private:
+        std::vector<std::uint8_t> palette;  /**< Vector containing colour data for this image */
+        std::vector<std::uint8_t> data;     /**< Vector containing image data or palette offsets */
+
+        std::int32_t width;     /**< Width of this texture in pixels */
+        std::int32_t height;    /**< Height of this texture in pixels */
```

Comment:
[It turns out we don't!!!!](https://www.opengl.org/discussion_boards/showthread.php/131495-glTexImage2D-use-or-copy-pixel-data)

I'll move this to a local variable, and put the `pixel` struct inside `msbmp.cpp`, though when we get some more common types going (vertex, etc), we can probably reuse that.